src/instructor_clj/core.clj

```diff
@@ -37,26 +61,35 @@
                                                 "content" (schema->system-prompt response-schema)}
                                                {"role" "user"
                                                 "content" prompt}]
-                                  "temperature" temprature
+                                  "temperature" temperature
                                   "max_tokens" max-tokens})
-        body (try
-               (-> (http/post api-url {:headers headers
-                                       :body body})
-                   deref ;; Dereference the future
-                   :body
-                   (cc/parse-string true))
-               (catch JsonParseException _))
-        response (try
-                   (-> body
-                       :choices
-                       first
-                       :message
-                       :content
-                       (cc/parse-string true))
-                   (catch JsonParseException _))]
-    (if (m/validate response-schema response)
-      response
-      body)))
+        body (call-llm-api api-url headers body)
+        response (parse-generated-body body)]
+    (when (m/validate response-schema response)
+      response)))
+
+
+(defn instruct
+  "Attempts to obtain a valid response from the LLM based on the given prompt and schema,
+   retrying up to `max-retries` times if necessary."
+  [prompt response-schema & {:keys [api-key max-tokens model temperature max-retries]
+                             :or {max-tokens 4096
+                                  model "gpt-3.5-turbo"
+                                  temperature 0.7
+                                  max-retries 0}}]
+  (loop [retries-left max-retries]
+    (let [response (try (llm->response {:prompt prompt
+                                        :response-schema response-schema
+                                        :api-key api-key
+                                        :max-tokens max-tokens
+                                        :model model
+                                        :temperature temperature})
+                        (catch Throwable _))]
```

Comment:
General catch all would suppress API failures as well. 
As a side note, A good place to start using Tardigrade here.